jib-cli/src/main/java/com/google/cloud/tools/jib/cli/cli2/JibCli.java

```diff
@@ -0,0 +1,357 @@
+  private boolean allowInsecureRegistries;
+
+  @Option(
+      names = "--send-credentials-over-http",
+      description = "Allow jib to communicate with registries over https")
+  @SuppressWarnings("NullAway.Init") // initialized by picocli
+  private boolean sendCredentialsOverHttp;
+
+  @Option(
+      names = {"--credential-helper"},
+      paramLabel = "<credential-helper>",
+      description =
+          "Add a credential helper, either a path to the helper, or a suffix for an executable named `docker-credential-<suffix>` (repeatable)")
+  private List<String> credentialHelpers = new ArrayList<>();
+
+  @ArgGroup(exclusive = true)
+  @SuppressWarnings("NullAway.Init") // initialized by picocli
+  private UsernamePassword usernamePassword;
+
+  private static class UsernamePassword {
+    @ArgGroup(exclusive = false)
+    @SuppressWarnings("NullAway.Init") // initialized by picocli
+    SingleUsernamePassword single;
+
+    @ArgGroup(exclusive = false)
+    @SuppressWarnings("NullAway.Init") // initialized by picocli
+    MultiUsernamePassword multi;
+  }
+
+  private static class MultiUsernamePassword {
+    @ArgGroup(exclusive = false)
+    @SuppressWarnings("NullAway.Init") // initialized by picocli
+    ToUsernamePassword to;
+
+    @ArgGroup(exclusive = false)
+    @SuppressWarnings("NullAway.Init") // initialized by picocli
+    FromUsernamePassword from;
+  }
+
+  private static class SingleUsernamePassword {
+    @Option(
+        names = "--username",
+        required = true,
+        description = "username for communicating with target/base image registry")
+    @SuppressWarnings("NullAway.Init") // initialized by picocli
+    String username;
+
+    @Option(
+        names = "--password",
+        arity = "0..1",
+        required = true,
+        interactive = true,
+        description = "password for communicating with target/base image registry")
+    @SuppressWarnings("NullAway.Init") // initialized by picocli
+    String password;
+  }
+
+  private static class ToUsernamePassword {
+    @Option(
+        names = "--to-username",
+        required = true,
+        description = "username for communicating with target image registry")
+    @SuppressWarnings("NullAway.Init") // initialized by picocli
+    String username;
+
+    @Option(
+        names = "--to-password",
+        arity = "0..1",
+        interactive = true,
+        required = true,
+        description = "password for communicating with target image registry")
+    @SuppressWarnings("NullAway.Init") // initialized by picocli
+    String password;
+  }
+
+  private static class FromUsernamePassword {
+    @Option(
+        names = "--from-username",
+        required = true,
+        description = "username for communicating with base image registry")
+    @SuppressWarnings("NullAway.Init") // initialized by picocli
+    String username;
+
+    @Option(
+        names = "--from-password",
+        arity = "0..1",
```

Comment:
QQ: what is this?

Reply:
arity is how many args this option can have.
```
--from-password (with no args) <-- triggers interactive flow
--password <password>
```
A little more info here: https://picocli.info/#_interactive_password_options

Reply:
Oh, it's nice that they have an interactive password option.